eZ/Bundle/EzPublishCoreBundle/Tests/Converter/RepositoryParamConverterTest.php

```diff
@@ -0,0 +1,202 @@
+
+        $this->setExpectedException( 'Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException', 'Access to values denied' );
+        $this->converter->apply( $request, $config );
+    }
+
+    public function testApplyContentNotFound404Exception()
+    {
+        $id = 42;
+        $request = new Request( array(), array(), array( RepositoryParamConverter::CONTENT_ID => $id ) );
+        $config = $this->createConfiguration( RepositoryParamConverter::CONTENT_CLASS, 'content' );
+
+        $this->contentServiceMock
+            ->expects( $this->once() )
+            ->method( 'loadContent' )
+            ->with( $id )
+            ->will( $this->throwException( new NotFoundException( '', '' ) ) );
+
+        $this->setExpectedException( 'Symfony\Component\HttpKernel\Exception\NotFoundHttpException', 'Requested values not found' );
+        $this->converter->apply( $request, $config );
+    }
+
+    public function testApplyContentUnauthorized404Exception()
+    {
+        $id = 42;
+        $request = new Request( array(), array(), array( RepositoryParamConverter::CONTENT_ID => $id ) );
+        $config = $this->createConfiguration( RepositoryParamConverter::CONTENT_CLASS, 'content' );
+
+        $this->contentServiceMock
+            ->expects( $this->once() )
+            ->method( 'loadContent' )
+            ->with( $id )
+            ->will( $this->throwException( new UnauthorizedException( '', '' ) ) );
+
+        $this->setExpectedException( 'Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException', 'Access to values denied' );
+        $this->converter->apply( $request, $config );
+    }
+
+    public function testApplyLocationOptionalWithEmptyAttribute()
+    {
+        $request = new Request( array(), array(), array( RepositoryParamConverter::LOCATION_ID => null ) );
+        $config = $this->createConfiguration( RepositoryParamConverter::LOCATION_CLASS, 'location' );
+
+        $config->expects( $this->once() )
+            ->method( 'isOptional' )
+            ->will( $this->returnValue( true ) );
+
+        $this->assertFalse( $this->converter->apply( $request, $config ) );
+        $this->assertNull( $request->attributes->get( 'location' ) );
+    }
+
+    public function testApplyContentOptionalWithEmptyAttribute()
+    {
+        $request = new Request( array(), array(), array( RepositoryParamConverter::CONTENT_ID => null ) );
+        $config = $this->createConfiguration( RepositoryParamConverter::CONTENT_CLASS, 'content' );
+
+        $config->expects( $this->once() )
+            ->method( 'isOptional' )
+            ->will( $this->returnValue( true ) );
+
+        $this->assertFalse( $this->converter->apply( $request, $config ) );
+        $this->assertNull( $request->attributes->get( 'content' ) );
+    }
+
+    public function createConfiguration($class = null, $name = null)
+    {
+        $config = $this
+            ->getMockBuilder( 'Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter' )
```

Comment:
First, to be honest, it's a copy/paste/modification from Sensio's tests on another converter.
Then, I don't think there is an interface for this (the configuration).